### PR TITLE
Home task 4

### PR DESCRIPTION
**Pull request by antonfilichkin**

### CODE REVIEW

File: ht4/hw1.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+class Homework:
+    def __init__(self, name: str, deadline: int):
+        self._name = name
+        self._deadline = deadline
+
+    def check_deadline(self):
+        if self._deadline < 0:
+            raise DeadlineError("You are late")
```
Comment (alpakhomov) [inline ht4/hw1.py:43]: Deadline should be a timedelta object https://docs.python.org/3/library/datetime.html#timedelta-objects

We can "remember" when the homework was creted in create hw method. Then, we will check if (creation_date + deadline) >= now

**Review by alpakhomov — CHANGES_REQUESTED** [review]
Please fix deadline type and related methods

File: ht4/hw1.py (hunk 1 above)
Comment (antonfilichkin) [reply ht4/hw1.py:43]: Fixed. 
But we could probably just store the deadline as a date then?
something like:
`self.deadline = date.today() + timedelta(days=deadline)`
And just compare the dates?

Comment (alpakhomov) [reply ht4/hw1.py:43]: Yep, we can, that's what I meant (sorry if confused you with my description). Doesn't matter, I just wanted you to take a look at datetime objects in python

**Review by alpakhomov — APPROVED** [review]
Approved!

**Review by MChislov — APPROVED** [review]

**Pull request merged**